app/src/main/java/com/example/mob_lab1/LoginActivity.java

```diff
@@ -0,0 +1,124 @@
+package com.example.mob_lab1;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+public class LoginActivity extends AppCompatActivity {
+
+    EditText emailID;
+    EditText password;
+    Button btnSignUp;
+    TextView tvSignIn;
+    FirebaseAuth mFirebaseAuth;
+    private int passMinLenght = 8;
+    public String emailPattern = "[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]+";
+
+    private FirebaseAuth.AuthStateListener mAuthStateListener;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+
+
+        mFirebaseAuth = FirebaseAuth.getInstance();
+        emailID = findViewById(R.id.editText);
+        password = findViewById(R.id.editText3);
+        btnSignUp = findViewById(R.id.button1);
+        tvSignIn = findViewById(R.id.textView);
+
+        mAuthStateListener = new FirebaseAuth.AuthStateListener() {
+            @Override
+            public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
+                FirebaseUser mFirebaseUser = mFirebaseAuth.getCurrentUser();
+                if(mFirebaseUser != null){
+                    Toast.makeText(LoginActivity.this,"You are logged in",Toast.LENGTH_SHORT).show();
+                    Intent i = new Intent(LoginActivity.this, HomeActivity.class);
+                    startActivity(i);
+                }
+                else {
+                    Toast.makeText(LoginActivity.this,"Please Login",Toast.LENGTH_SHORT).show();
+                }
+            }
+        };
+
+        btnSignUp.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String email = emailID.getText().toString();
+                String pwd = password.getText().toString();
+
+                if(email.isEmpty()){
+                    emailID.setError("Please enter your email");
+                    emailID.requestFocus();
+                }
+                else if(pwd.isEmpty()) {
+                    password.setError("Please enter your password");
+                    password.requestFocus();
+                }
+                else if(password.length() < passMinLenght){  // Validation of pass
+                    password.setText("");

```

Comment:
так не варто очищати поля
```suggestion
                    password.getText().clear(); 
```